terasoluna-gfw-common-libraries/terasoluna-gfw-common/src/main/java/org/terasoluna/gfw/common/codelist/i18n/SimpleI18nCodeList.java

```diff
@@ -226,14 +232,54 @@
 
     /**
      * <p>
-     * returns row of codelist table.
+     * Returns a codelist as map for the {@link org.springframework.context.i18n.LocaleContextHolder}'s locale.
+     * </p>
+     * @see org.terasoluna.gfw.common.codelist.CodeList#asMap()
+     */
+    @Override
+    public Map<String, String> asMap() {
+        return asMap(LocaleContextHolder.getLocale());
+    }
+
+    /**
+     * <p>
+     * returns row of codelist table.<br>
+     * if there is no codelist for the locale, throws an {@link IllegalStateException}.
      * </p>
      * @see org.terasoluna.gfw.common.codelist.i18n.I18nCodeList#asMap(java.util.Locale)
      */
     @Override
     public Map<String, String> asMap(Locale locale) {
         Assert.notNull(locale, "locale is null");
-        return codeListTable.row(locale);
+
+        if (codeListTable.containsRow(locale)) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Found codelist for specified locale '{}'.",
+                        locale);
+            }
+            return codeListTable.row(locale);
+        }
```

Comment:
- [ ] I think that resolving the locale and getting the CodeList should separate and be able to override resolving the locale.

For example::
```java
public Map<String, String> asMap(Locale locale) {
  Assert.notNull(locale, "locale is null";
  Locale codeListLocale = resolveCodeListLocale(locale);
  return codeListTable.row(codeListLocale);
}

protected Locale resolveCodeListLocale(Locale locale) {
  if (codeListTable.containsRow(locale)) {
    logger.debug("Found codelist for specified locale '{}'.", locale);
    return locale;
  }
  if (locale.getCountry().length() > 0) {
    Locale langOnlyLocale = new Locale(locale.getLanguage());
    if (codeListTable.containsRow(langOnlyLocale)) {
      logger.debug("No codelist for specified locale '{}'. Use '{}' as language only.", locale, langOnlyLocale);
      return langOnlyLocale;
    }
  }
  logger.debug("No codelist for specified locale '{}'. Use '{}' as fallback.", locale, fallbackTo);
  return fallbackTo;
}
```

Reply:
Previously I separated, but finally I mergerd.
I forgot that reason, so I will reconsider it.

Reply:
It may be better to separate language locale resolution and fallback.

Reply:
Modified by 05ea3f1